src/main/java/net/honux/chess/pieces/Piece.java

```diff
@@ -0,0 +1,95 @@
+    public static final String BLACK_COLOR = "black";
+
+    public static final char WHITE_PAWN_REPRESENTATION = 'p';
+    public static final char BLACK_PAWN_REPRESENTATION = 'P';
+    public static final char WHITE_KNIGHT_REPRESENTATION = 'n';
+    public static final char BLACK_KNIGHT_REPRESENTATION = 'N';
+    public static final char WHITE_ROOK_REPRESENTATION = 'r';
+    public static final char BLACK_ROOK_REPRESENTATION = 'R';
+    public static final char WHITE_BISHOP_REPRESENTATION = 'b';
+    public static final char BLACK_BISHOP_REPRESENTATION = 'B';
+    public static final char WHITE_QUEEN_REPRESENTATION = 'q';
+    public static final char BLACK_QUEEN_REPRESENTATION = 'Q';
+    public static final char WHITE_KING_REPRESENTATION = 'k';
+    public static final char BLACK_KING_REPRESENTATION = 'K';
+
+
+    private String color;
+    private char representation;
+
+
+    private Piece(String color, char representation) {
+        this.color = color;
+        this.representation = representation;
+    }
+
+    public static Piece createWhitePawn() {
+        return new Piece(Piece.WHITE_COLOR, Piece.WHITE_PAWN_REPRESENTATION);
+    }
+
+    public static Piece createBlackPawn() {
+        return new Piece(Piece.BLACK_COLOR, Piece.BLACK_PAWN_REPRESENTATION);
+    }
+
+    public static Piece createWhiteKnight() {
+        return new Piece(Piece.WHITE_COLOR, Piece.WHITE_KNIGHT_REPRESENTATION);
+    }
+
+    public static Piece createBlackKnight() {
+        return new Piece(Piece.BLACK_COLOR, Piece.BLACK_KNIGHT_REPRESENTATION);
+    }
+
+    public static Piece createWhiteRook() {
+        return new Piece(Piece.WHITE_COLOR, Piece.WHITE_ROOK_REPRESENTATION);
+    }
+
+    public static Piece createBlackRook() {
+        return new Piece(Piece.BLACK_COLOR, Piece.BLACK_ROOK_REPRESENTATION);
+    }
+
+    public static Piece createWhiteBishop() {
+        return new Piece(Piece.WHITE_COLOR, Piece.WHITE_BISHOP_REPRESENTATION);
+    }
+
+    public static Piece createBlackBishop() {
+        return new Piece(Piece.BLACK_COLOR, Piece.BLACK_BISHOP_REPRESENTATION);
+    }
+
+    public static Piece createWhiteQueen() {
+        return new Piece(Piece.WHITE_COLOR, Piece.WHITE_QUEEN_REPRESENTATION);
+    }
+
+    public static Piece createBlackQueen() {
+        return new Piece(Piece.BLACK_COLOR, Piece.BLACK_QUEEN_REPRESENTATION);
+    }
+
+    public static Piece createWhiteKing() {
+        return new Piece(Piece.WHITE_COLOR, Piece.WHITE_KING_REPRESENTATION);
+    }
+
+    public static Piece createBlackKing() {
+        return new Piece(Piece.BLACK_COLOR, Piece.BLACK_KING_REPRESENTATION);
+    }
+
+
+    public String getColor() {
+        return color;
+    }
+
+    public char getRepresentation() {
+        return representation;
+    }
+
+    public boolean isWhite(){
+        return this.color.equals(WHITE_COLOR);
+    }
+
+    public boolean isBlack(){
+        return this.color.equals(BLACK_COLOR);
+    }
```

Comment:
```suggestion
    public boolean isWhite() {
        return WHITE_COLOR.equals(this.color);
    }

    public boolean isBlack() {
        return BLACK_COLOR.equals(this.color);
    }
```

왜 이 코드가 더 안전할까요? 한 번 생각해보세요.

Reply:
이렇게 해야 들어오는 color 값으로 맞는지 아닌지 검증을 할 수 있으니까~? 일까요? 그리고 디온과 같이 진행을 하게 되면 null이 들어오는 경우도, try로 잡아줘야 할까요?

Reply:
이렇게 들어오면, null 처리를 해주지 않아도 됩니다.
그 이유는 직접 한 번 해보세요~